src/audio/module_adapter/module/volume/volume_generic.c

```diff
@@ -93,6 +93,47 @@ static void vol_s24_to_s24(struct processing_module *mod, struct input_stream_bu
 		y = audio_stream_wrap(sink, y + n);
 	}
 }
+
+/**
+ * \brief Volume bypass from 24/32 bit to 24/32 bit.
+ * \param[in,out] dev Volume base component device.
+ * \param[in,out] sink Destination buffer.
+ * \param[in,out] source Input buffer.
+ * \param[in] frames Number of frames to process.
+ * \param[in] attenuation factor for peakmeter adjustment (unused)
+ *
+ * Copy and scale volume from 24/32 bit source buffer
+ * to 24/32 bit destination buffer.
+ */
+static void vol_bypass_s24_to_s24(struct processing_module *mod,
+				  struct input_stream_buffer *bsource,
+				  struct output_stream_buffer *bsink, uint32_t frames,
+				  uint32_t attenuation)
+{
+	struct audio_stream __sparse_cache *source = bsource->data;
+	struct audio_stream __sparse_cache *sink = bsink->data;
+	int32_t *x;
+	int32_t *y;
+	int nmax, n;
+	const int nch = audio_stream_get_channels(source);
+	int remaining_samples = frames * nch;
+
+	x = audio_stream_wrap(source, (char *)audio_stream_get_rptr(source) + bsource->consumed);
```

Comment:
Use audio_stream_copy()

Reply:
not so suitable since when ramp, we won't update the read/ write pointer inside the while loop of volume_process.